File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <algorithm>
#include <vector>

using namespace std;

int solution(vector<int> elements) {
  int size = elements.size();
  vector<int> consecutive_sums;

  for (int i = 0; i < size; i++) elements.push_back(elements[i]);
  for (int i = 0; i < size; i++) {
    int sum = 0;

    for (int j = 0; j < size; j++) {
      sum += elements[i + j];
      consecutive_sums.push_back(sum);
    }
  }
  sort(consecutive_sums.begin(), consecutive_sums.end());
  consecutive_sums.erase(
      unique(consecutive_sums.begin(), consecutive_sums.end()),
      consecutive_sums.end());
  return consecutive_sums.size();
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.HashSet;
import java.util.Set;

class Solution {

    public int solution(int[] elements) {
        int size = elements.length;
        int[] extendedElements = new int[2 * size];
        Set<Integer> consecutiveSums = new HashSet<>();

        System.arraycopy(elements, 0, extendedElements, 0, size);
        System.arraycopy(elements, 0, extendedElements, size, size);
        for (int i = 0; i < size; i++) {
            int sum = 0;

            for (int j = 0; j < size; j++) {
                sum += extendedElements[i + j];
                consecutiveSums.add(sum);
            }
        }
        return consecutiveSums.size();
    }
}